src/sync/chase_lev.rs

```diff
@@ -254,6 +273,40 @@ impl<T> Deque<T> {
         }
     }
 
+    fn steal_half(&self) -> StealHalf<T> {
+        let guard = epoch::pin();
+
+        let t = self.top.load(Acquire);
+        fence(SeqCst);
+        let b = self.bottom.load(Acquire);
+
+        let size = b - t;
+        if size <= 0 {
+            return StealHalf::Empty;
+        }
+
+        let half = size + 1 / 2;
```

Comment:
I think this is indeed a bug.  The stealer effectively tries to steal all the remaining elements, breaking the invariant for the popper that the last element will not be stolen.  I think it should be `let half = (size + 1) / 2;`